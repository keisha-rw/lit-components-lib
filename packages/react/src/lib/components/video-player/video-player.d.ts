import * as React from 'react';
import { PdsVideoPlayerProps } from '../../built-component-props';
import '../../../../i18n';
import '@keisha/design-system/css/video-player/video-player-light-dom.scss';
export declare const PdsVideoPlayer: React.ForwardRefExoticComponent<Omit<PdsVideoPlayerProps, "ref"> & React.RefAttributes<React.Ref<any>>>;
