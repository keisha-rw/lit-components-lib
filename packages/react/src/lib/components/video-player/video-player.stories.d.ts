import * as React from 'react';
import { PdsVideoPlayerProps } from '../../built-component-props';
declare const _default: {
    title: string;
    component: React.ForwardRefExoticComponent<Omit<PdsVideoPlayerProps, "ref"> & React.RefAttributes<React.Ref<any>>>;
    tags: string[];
    parameters: {
        happo: boolean;
    };
};
export default _default;
export declare const Default: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Omit<PdsVideoPlayerProps, "ref"> & React.RefAttributes<React.Ref<any>>>;
export declare const Alt: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0fc72a6d").R, Omit<PdsVideoPlayerProps, "ref"> & React.RefAttributes<React.Ref<any>>>;
