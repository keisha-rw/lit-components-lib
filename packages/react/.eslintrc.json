{
  "extends": ["../../.eslintrc.json"],
  "ignorePatterns": [
    "!**/*",
    "node_modules",
    "*.d.ts"
  ],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {
        "import/no-extraneous-dependencies": "off",
        "import/prefer-default-export": "off",
        "curly": ["error", "all"],
        "react/function-component-definition": [
          2,
          {
            "namedComponents": "arrow-function",
            "unnamedComponents": "arrow-function"
          }
        ],
        "react/jsx-props-no-spreading": [
          "error",
          {
            "custom": "ignore"
          }
        ]
      },
      "extends": ["@keisha/eslint-config-react-typescript"],
      "parserOptions": {
        "project": ["packages/react/tsconfig.lib.json"]
      }
    },
    {
      "files": ["*.spec.tsx"],
      "rules": {
        "import/no-extraneous-dependencies": "off",
        "import/prefer-default-export": "off",
        "curly": ["error", "all"],
        "react/function-component-definition": [
          2,
          {
            "namedComponents": "arrow-function",
            "unnamedComponents": "arrow-function"
          }
        ],
        "react/jsx-props-no-spreading": [
          "error",
          {
            "custom": "ignore"
          }
        ]
      },
      "extends": ["@keisha/eslint-config-react-typescript"],
      "parserOptions": {
        "project": ["packages/react/tsconfig.spec.json"]
      }
    },
    {
      "files": ["*.stories.tsx"],
      "parserOptions": {
        "project": ["packages/react/.storybook/tsconfig.json"]
      },
      "rules": {
        "import/no-extraneous-dependencies": "off",
        "import/prefer-default-export": "off",
        "curly": ["error", "all"],
        "react/function-component-definition": "off",
        "react/jsx-props-no-spreading": [
          "error",
          {
            "custom": "ignore"
          }
        ]
      },
      "extends": ["@keisha/eslint-config-react-typescript"]
    },
    {
      "files": ["*.js", "*.jsx"],
      "extends": ["@keisha/eslint-config-react-typescript"],
      "env": {
        "es6": true
      },
      "parserOptions": {
        "project": ["packages/react/tsconfig.lib.json"],
        "ecmaVersion": 9,
        "sourceType": "module"
      },
      "rules": {
        "import/no-extraneous-dependencies": "off",
        "import/prefer-default-export": "off",
        "curly": ["error", "all"],
        "react/jsx-props-no-spreading": [
          "error",
          {
            "custom": "ignore"
          }
        ]
      }
    },
    {
      "files": [".storybook/*.js"],
      "rules": {},
      "parserOptions": {
        "sourceType": "module",
        "ecmaVersion": 2018,
        "project": ["packages/react/.storybook/tsconfig.json"]
      }
    },
    {
      "files": ["jest.config.ts"],
      "parserOptions": {
        "project": ["packages/react/tsconfig.spec.json"]
      },
      "extends": ["@keisha/eslint-config-typescript"],
      "rules": {
        "curly": ["error", "all"]
      }
    }
  ]
}
