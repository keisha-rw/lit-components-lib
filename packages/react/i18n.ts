import i18next from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

// TODO: move the translations to JSON files
// TODO: see if we can parse the xlf file to generate the JSON files

i18next
  .use(initReactI18next)
  .use(LanguageDetector)
  .init({
    fallbackLng: 'en',
    resources: {
      en: {
        translation: {
          'video-duration-is': 'Video duration is',
          'minutes-and': 'minutes and',
          'seconds-long': 'seconds long',
          'view-video-transcript': 'View video transcript',
        },
      },
      es: {
        translation: {
          'video-duration-is': 'La duración del vídeo es de',
          'minutes-and': 'minutos y',
          'seconds-long': 'segundos',
          'view-video-transcript': 'Ver la transcripción del video',
        },
      },
    },
    detection: {
      lookupLocalStorage: 'pds-language',
      lookupQuerystring: 'lang',
      order: [
        'localStorage',
        'htmlTag',
        'querystring',
        'cookie',
        'sessionStorage',
        'navigator',
      ],
    },
  });

i18next.languages = ['en', 'es'];

export default i18next;
