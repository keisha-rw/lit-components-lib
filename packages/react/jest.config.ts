/* eslint-disable */
const esModules = [
  'chai-a11y-axe',
  'cleave.js',
  'lit',
  'lit-html',
  '@lit',
  '@open-wc',
  '@web',
  '@esm-bundle',
  'storybook-mock-date-decorator',
].join('|');

export default {
  displayName: 'react',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\.[tj]sx?$': ['ts-jest', {
      tsconfig: '<rootDir>/tsconfig.spec.json',
    }],
    '^.+\.mjs$': ['babel-jest', { presets: ['@nrwl/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/packages/react',
  collectCoverage: true,
  collectCoverageFrom: [
    '<rootDir>/src/lib/components/video-player/video-player.tsx',
  ],
  coveragePathIgnorePatterns: ['/node_modules/'],
  transformIgnorePatterns: [`/node_modules/(?!${esModules})`, 'packages/web-components/src/lib/components/.+.ts'],
  coverageReporters: ['clover', 'json', 'lcov', 'text'],
  moduleNameMapper: {
    'cleave.js': '<rootDir>/../../node_modules/cleave.js/dist/cleave-esm.js',
    '@keisha/design-system-icons-react': '<rootDir>/../../dist/packages/icons-react/cjs/index.js',
    '@keisha/design-system-icons-web': '<rootDir>/../../dist/packages/icons-web/cjs/index.js',
    '@keisha/design-system/utils/video-player/video-player-utils': '<rootDir>/../web-components/src/lib/components/video-player/video-player-utils-mock.ts',
    '@keisha/design-system': '<rootDir>/../../dist/packages/web-components/cjs/index.js',
    '.+\\.(css|styl|less|sass|scss|svg|png|jpg|ttf|woff|woff2)(\\?inline)?$': 'identity-obj-proxy',
  },
  coverageThreshold: {
    global: {
      branches: 38,
      functions: 45,
      lines: 67,
      statements: 67,
    },
  },
  // TODO: uncomment this later
  // coverageThreshold: {
  //   global: {
  //     branches: 90,
  //     functions: 90,
  //     lines: 90,
  //     statements: 90,
  //   },
  //   './packages/react/src/lib/components/video-player/video-player.tsx': {
  //     branches: 90,
  //     functions: 90,
  //     lines: 90,
  //     statements: 90,
  //   },
  // },
};
