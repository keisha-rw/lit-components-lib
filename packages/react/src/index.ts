// eslint-disable-next-line import/extensions
import '@lit-labs/ssr-react/enable-lit-ssr.js';

export * from './lib/components/action-menu-item/action-menu-item';
export * from './lib/components/action-menu/action-menu';
export * from './lib/components/alert/alert';
export * from './lib/components/avatar/avatar';
export * from './lib/components/band/band';
export * from './lib/components/box-button/box-button';
export * from './lib/components/breadcrumbs-item/breadcrumbs-item';
export * from './lib/components/breadcrumbs/breadcrumbs';
export * from './lib/components/button/button';
export * from './lib/components/card/card';
export * from './lib/components/checkbox/checkbox';
export * from './lib/components/collapsible/collapsible';
export * from './lib/components/data-table-cell/data-table-cell';
export * from './lib/components/data-table-column/data-table-column';
export * from './lib/components/data-table-columns/data-table-columns';
export * from './lib/components/data-table-input-column/data-table-input-column';
export * from './lib/components/data-table-row/data-table-row';
export * from './lib/components/data-table-rows/data-table-rows';
export * from './lib/components/data-table/data-table';
export * from './lib/components/date-picker-input/date-picker-input';
export * from './lib/components/date-picker/date-picker';
export * from './lib/components/decorator/decorator';
export * from './lib/components/error-page/error-page';
export * from './lib/components/feature-block/feature-block';
export * from './lib/components/file-upload/file-upload';
export * from './lib/components/footer-contact-link/footer-contact-link';
export * from './lib/components/footer-nav-item/footer-nav-item';
export * from './lib/components/footer-nav/footer-nav';
export * from './lib/components/footer/footer';
export * from './lib/components/footnote-item/footnote-item';
export * from './lib/components/footnote-link/footnote-link';
export * from './lib/components/footnote/footnote';
export * from './lib/components/grid-item/grid-item';
export * from './lib/components/grid/grid';
export * from './lib/components/heading/heading';
export * from './lib/components/hr/hr';
export * from './lib/components/layout-container/layout-container';
export * from './lib/components/linelength-container/linelength-container';
export * from './lib/components/link/link';
export * from './lib/components/list-item/list-item';
export * from './lib/components/list/list';
export * from './lib/components/loading-page/loading-page';
export * from './lib/components/logo/logo';
export * from './lib/components/logout-page/logout-page';
export * from './lib/components/modal/modal';
export * from './lib/components/pagination-item/pagination-item';
export * from './lib/components/pagination/pagination';
export * from './lib/components/phone-number-input/phone-number-input';
export * from './lib/components/primary-navigation-container/primary-navigation-container';
export * from './lib/components/primary-navigation-dropdown-link/primary-navigation-dropdown-link';
export * from './lib/components/primary-navigation-main-menu-item/primary-navigation-main-menu-item';
export * from './lib/components/primary-navigation-main-menu/primary-navigation-main-menu';
export * from './lib/components/primary-navigation-utility-menu-item/primary-navigation-utility-menu-item';
export * from './lib/components/primary-navigation-utility-menu/primary-navigation-utility-menu';
export * from './lib/components/primary-navigation/primary-navigation';
export * from './lib/components/radio-group/radio-group';
export * from './lib/components/radio/radio';
export * from './lib/components/secondary-navigation-level-one/secondary-navigation-level-one';
export * from './lib/components/secondary-navigation-level-two/secondary-navigation-level-two';
export * from './lib/components/secondary-navigation-link/secondary-navigation-link';
export * from './lib/components/secondary-navigation/secondary-navigation';
export * from './lib/components/segmented-control-item/segmented-control-item';
export * from './lib/components/segmented-control/segmented-control';
export * from './lib/components/select/select';
export * from './lib/components/session-timer-countdown/session-timer-countdown';
export * from './lib/components/session-timer/session-timer';
export * from './lib/components/show-more/show-more';
export * from './lib/components/sidebar/sidebar';
export * from './lib/components/skeleton-loader/skeleton-loader';
export * from './lib/components/status-indicator/status-indicator';
export * from './lib/components/step-indicator-item/step-indicator-item';
export * from './lib/components/step-indicator/step-indicator';
export * from './lib/components/support-heading/support-heading';
export * from './lib/components/switch/switch';
export * from './lib/components/table/table';
export * from './lib/components/tag/tag';
export * from './lib/components/text-area/text-area';
export * from './lib/components/text-input/text-input';
export * from './lib/components/text-passage/text-passage';
export * from './lib/components/tooltip/tooltip';
export * from './lib/components/video-player/video-player';
