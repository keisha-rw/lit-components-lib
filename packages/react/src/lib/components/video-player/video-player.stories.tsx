import * as React from 'react';
import { StoryFn } from '@storybook/react';
import { PdsVideoPlayer } from './video-player';
import { PdsVideoPlayerProps } from '../../built-component-props';

export default {
  title: 'Components/Video player',
  component: PdsVideoPlayer,
  tags: ['autodocs'],
  parameters: { happo: false },
};

const Template: StoryFn<typeof PdsVideoPlayer> = (
  args: PdsVideoPlayerProps,
) => (
  <PdsVideoPlayer
    {...args}
    videoUrl="https://players.brightcove.net/1162425107001/cbe63a49-d26e-41ed-9e8e-5c1dec8c6b45_default/index.html?videoId=5027867735001"
    variant="alt"
    videoPlayerTextTracks="https://static.3playmedia.com/p/files/5281753/threeplay_transcripts/15765393?format_id=16&amp;project_id=17743&amp;format=txt"
  />
);

export const Default = Template.bind({});
Default.args = {
  videoTitle: 'This is an optional video title',
  videoDescription:
    '<p>This is an optional <strong>video description</strong>. It should only be added if there is also a video title.</p>',
};

Default.parameters = {
  actions: {
    handles: [
      'pds-video-player-play',
      'pds-video-player-ended',
      'pds-video-player-progress',
    ],
  },
};

export const Alt = Template.bind({});
Alt.args = {
  variant: 'alt',
  hideVideoDuration: true,
};

Alt.parameters = {
  actions: {
    handles: [
      'pds-video-player-play',
      'pds-video-player-ended',
      'pds-video-player-progress',
    ],
  },
};
