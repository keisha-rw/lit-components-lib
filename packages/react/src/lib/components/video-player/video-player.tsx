import * as React from 'react';
import { createRoot } from 'react-dom/client';
import ReactPlayerLoader from '@brightcove/react-player-loader';
import { PdsIconClock } from '@keisha/design-system-icons-react';
import { PdsVideoPlayer as PdsVideoPlayerWebComponent } from '@keisha/design-system';
import { useTranslation } from 'react-i18next';
import sanitizeHTML from 'sanitize-html';
import { ALLOWED_TAGS_LIST } from '@keisha/design-system/utils/video-player/video-player-utils';
import { v4 as uuidv4 } from 'uuid';
import { PdsLink } from '../link/link';
import { PdsVideoPlayerProps } from '../../built-component-props';
import '../../../../i18n';
import { PdsTextPassage } from '../text-passage/text-passage';
import '@keisha/design-system/css/video-player/video-player-light-dom.scss';

export const PdsVideoPlayer = React.forwardRef<
  React.Ref<any>,
  PdsVideoPlayerProps
>((props: PdsVideoPlayerProps, ref?: React.Ref<any>) => {
  const { t } = useTranslation();
  const {
    videoUrl,
    aspectRatio = '16:9',
    videoTitle,
    hideVideoDuration = false,
    videoPlayerTextTracks,
    videoDescription,
  } = props;
  const sanitizedVideoDescription = sanitizeHTML(videoDescription!, {
    allowedTags: ALLOWED_TAGS_LIST,
  });
  const [brightcoveVideoTitle, setBrightcoveVideoTitle] = React.useState<
    string | undefined
  >(undefined);
  const [videoDuration, setVideoDuration] = React.useState<string | undefined>(
    undefined,
  );
  const [minutes, setMinutes] = React.useState<string>('');
  const [seconds, setSeconds] = React.useState<string>('');
  const [percentage, setPercentage] = React.useState<string>('25%');
  const uniqueIdForVideo = React.useMemo(() => {
    return uuidv4();
  }, []);
  const videoPlayerWebComponent = React.useMemo(() => {
    return new PdsVideoPlayerWebComponent();
  }, []);
  const { accountId, playerId, videoId } = React.useMemo(() => {
    return videoPlayerWebComponent.retrieveVideoInformationFromUrl(videoUrl);
  }, [videoUrl, videoPlayerWebComponent]);

  const onFailure = (error: any) => {
    // eslint-disable-next-line no-console
    console.error(
      'PDS video player encountered an error: ',
      JSON.stringify(error),
    );
  };

  React.useEffect(() => {
    const wrapperDiv = document.getElementById(
      `pds-video-player-container-${videoId}-${uniqueIdForVideo}`,
    );

    const onSuccess = (success: any) => {
      success.ref.on(
        'loadstart',
        (player: {
          target: { player: { mediainfo: { duration: string; name: string } } };
        }) => {
          const {
            brightcoveVideoTitle: videoTitleFromComponent,
            minutes: minutesFromComponent,
            seconds: secondsFromComponent,
            videoDuration: videoDurationFromComponent,
          } = videoPlayerWebComponent.handleLoadstart({
            player,
            hideVideoDuration,
          });
          setBrightcoveVideoTitle(videoTitleFromComponent);
          setMinutes(minutesFromComponent);
          setSeconds(secondsFromComponent);
          setVideoDuration(videoDurationFromComponent);
        },
      );

      /**
       * Utilizes the Brightcove play event to dispatch a custom event
       * Used for GA4 tracking for when a user plays a video
       */
      success.ref.on('play', () => {
        videoPlayerWebComponent.handlePlay({
          videoUrl,
          elementToFireEventFrom: wrapperDiv!,
        });
      });

      /**
       * Utilizes the Brightcove ended event to dispatch a custom event
       * Used for GA4 tracking for when a user gets to the end of a video
       */
      success.ref.on('ended', () => {
        videoPlayerWebComponent.handleEnded({
          videoUrl,
          elementToFireEventFrom: wrapperDiv!,
        });
      });

      /**
       * Utilizes the Brightcove timeupdate event to dispatch a custom event
       * Used for GA4 tracking of when a user gets to 25%, 50%, 75%, and 90% of a video
       */
      success.ref.on(
        'timeupdate',
        (player: {
          target: {
            player: {
              controlBar: { currentTimeDisplay: { formattedTime_: string } };
            };
          };
        }) => {
          videoPlayerWebComponent.handleTimeupdate({
            player,
            elementToFireEventFrom: wrapperDiv!,
            videoUrl,
            percentage,
            percentageSetterFunction: setPercentage,
            seconds,
            minutes,
          });
        },
      );

      const videoPlayerElement = wrapperDiv?.querySelector(
        '.pds-c-video-player',
      );
      const videoPlayerDescription = wrapperDiv?.querySelector(
        '.pds-c-video-player__description',
      );

      if (videoPlayerElement && videoPlayerDescription) {
        videoPlayerElement?.setAttribute(
          'aria-describedby',
          `videoplayerdescription-${videoId}-${uniqueIdForVideo}`,
        );
        videoPlayerDescription.id = `videoplayerdescription-${videoId}-${uniqueIdForVideo}`;
      }
    };

    if (typeof window !== 'undefined') {
      const playerDiv = document.getElementById(
        `pds-video-player-${videoId}-${uniqueIdForVideo}`,
      );
      // Note: this createRoot rigmarole is only because
      // it is required by the react-player-loader library
      const root = createRoot(playerDiv!);
      root.render(
        <ReactPlayerLoader
          accountId={accountId}
          playerId={playerId}
          videoId={videoId}
          onSuccess={onSuccess}
          onFailure={onFailure}
          options={{ aspectRatio }}
        />,
      );
    }
    // We only want to run this hook once, so we don't want to pass in all the deps
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <div
      ref={ref}
      id={`pds-video-player-container-${videoId}-${uniqueIdForVideo}`}
    >
      <div
        className="pds-c-video-player"
        title={videoTitle || brightcoveVideoTitle}
      >
        <div id={`pds-video-player-${videoId}-${uniqueIdForVideo}`} />
        <div className="pds-u-sr-only">
          {t('video-duration-is')} {minutes} {t('minutes-and')}
          {seconds} {t('seconds-long')}.
        </div>
        <p
          aria-hidden="true"
          className="pds-c-video-player__duration"
          data-happo-hide
        >
          {!hideVideoDuration && (
            <>
              <span className="pds-c-video-player__icon">
                <PdsIconClock size="xs" />
              </span>
              <span>{videoDuration}</span>
            </>
          )}
        </p>
        {videoTitle && (
          <div>
            <p aria-hidden="true" className="pds-c-video-player-video-title">
              {videoTitle}
            </p>
          </div>
        )}
        {sanitizedVideoDescription && (
          <div>
            <PdsTextPassage
              lineLength="none"
              className="pds-c-video-player__description"
            >
              <span
                // eslint-disable-next-line react/no-danger
                dangerouslySetInnerHTML={{ __html: sanitizedVideoDescription }}
              />
            </PdsTextPassage>
          </div>
        )}
        <PdsLink
          className="pds-c-video-player__link"
          href={videoPlayerTextTracks}
          target="_blank"
        >
          {t('view-video-transcript')}
        </PdsLink>
      </div>
    </div>
  );
});
