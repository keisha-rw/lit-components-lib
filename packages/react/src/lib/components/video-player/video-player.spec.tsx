import { prettyShadowDOM } from 'shadow-dom-testing-library';
import { render } from '@testing-library/react';
import React from 'react';
import { PdsVideoPlayer } from './video-player';

jest.mock(
  '@keisha/design-system/utils/video-player/video-player-utils',
  () => {
    return { ALLOWED_TAGS_LIST: [] };
  },
);

jest.mock('uuid', () => ({
  __esModule: true,
  v4: () => 'abc123nowihaveauuid',
}));

describe('VideoPlayer unit tests', () => {
  beforeEach(async () => {});

  test('should render the video player with the defaults', () => {
    const { baseElement } = render(
      <PdsVideoPlayer
        videoUrl="https://players.brightcove.net/1162425107001/cbe63a49-d26e-41ed-9e8e-5c1dec8c6b45_default/index.html?videoId=5027867735001"
        variant="alt"
        videoPlayerTextTracks="https://static.3playmedia.com/p/files/5281753/threeplay_transcripts/15765393?format_id=16&amp;project_id=17743&amp;format=txt"
        videoTitle="This is an optional video title"
        videoDescription="<p>This is an optional <strong>video description</strong>. It should only be added if there is also a video title.</p>"
      />,
    );

    expect(prettyShadowDOM(baseElement)).toMatchSnapshot();
  });
});
