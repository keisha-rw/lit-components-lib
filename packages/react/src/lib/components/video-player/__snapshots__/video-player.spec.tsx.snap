// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer unit tests should render the video player with the defaults 1`] = `
"<body>
  <div>
    <div
      id="pds-video-player-container-5027867735001-abc123nowihaveauuid"
    >
      <div
        class="pds-c-video-player"
        title="This is an optional video title"
      >
        <div
          id="pds-video-player-5027867735001-abc123nowihaveauuid"
        >
          <div
            class="brightcove-react-player-loader"
          >
            <video-js
              class="video-js"
              controls="controls"
              data-video-id="5027867735001"
            />
          </div>
        </div>
        <div
          class="pds-u-sr-only"
        >
          Video duration is
           
           
          minutes and
           
          seconds long
          .
        </div>
        <p
          aria-hidden="true"
          class="pds-c-video-player__duration"
          data-happo-hide="true"
        >
          <span
            class="pds-c-video-player__icon"
          >
            <pds-icon-clock>
              <ShadowRoot />
            </pds-icon-clock>
          </span>
          <span />
        </p>
        <div>
          <p
            aria-hidden="true"
            class="pds-c-video-player-video-title"
          >
            This is an optional video title
          </p>
        </div>
        <div>
          <pds-text-passage
            class="pds-c-video-player__description"
          >
            <ShadowRoot />
            <span>
              This is an optional video description. It should only be added if there is also a video title.
            </span>
          </pds-text-passage>
        </div>
        <pds-link
          class="pds-c-video-player__link"
        >
          <ShadowRoot />
          View video transcript
        </pds-link>
      </div>
    </div>
  </div>
</body>"
`;
